test/metabase/api/native_query_snippet_test.clj

```diff
@@ -0,0 +1,147 @@
+(ns metabase.api.native-query-snippet-test
+  "Tests for /api/native-query-snippet endpoints."
+  (:require [clojure
+             [string :as str]
+             [test :refer :all]]
+            [metabase.models
+             [database :refer [Database]]
+             [native-query-snippet :refer [NativeQuerySnippet]]
+             [permissions :as perms]
+             [permissions-group :as group]]
+            [metabase.test.data.users :refer [fetch-user user->client]]
+            [toucan.util.test :as tt])
+  (:import java.time.LocalDateTime))
+
+(def ^:private test-snippet-fields [:content :creator_id :database_id :description :name])
+(def ^:private rasta (fetch-user :rasta))
+
+(defn- snippet-url
+  [& [arg & _]]
+  (str "native-query-snippet"
+       (when arg (str "/" arg))))
+
+(defn- name-schema-error?
+  [response]
+  (str/starts-with? (or (get-in response [:errors :name]) "")
+                    "Value does not match schema: "))
+
+;; GET /api/native-query-snippet
+(deftest list-snippets-api-test
+  (tt/with-temp* [Database           [db]
```

Comment:
almost everything important for writing tests is in the `metabase.test` namespace. So you can require that and use `mt/` instead of `tt/`, `test-users/`, etc.

Reply:
Noted, thanks. Updated all test files touched in this PR.